src/hooks/routing/useRouterTrade.ts

```diff
@@ -41,50 +42,49 @@ export function useRouterTrade(
   const { provider } = useWeb3React()
   const queryArgs = useGetQuoteArgs({ provider, tradeType, amountSpecified, otherCurrency, routerUrl })
 
+  // PRICE fetching is informational and costly, so it is done less frequently.
+  const pollingInterval = routerPreference === RouterPreference.PRICE ? ms`2m` : ms`15s`
+
   // Get the cached state *immediately* to update the UI without sending a request - using useGetQuoteQueryState -
   // but debounce the actual request - using useLazyGetQuoteQuery - to avoid flooding the router / JSON-RPC endpoints.
-  const { isError, data, currentData, fulfilledTimeStamp } = useGetQuoteQueryState(queryArgs)
-  const isValidBlock = useIsValidBlock(Number(data?.blockNumber))
-  const isSyncing = currentData !== data || !isValidBlock
+  const { isError, isFetching, data, currentData, fulfilledTimeStamp } = useGetQuoteQueryState(queryArgs)
 
-  // PRICE fetching is informational and costly, so it's done less frequently.
-  const pollingInterval = routerPreference === RouterPreference.PRICE ? ms`2m` : ms`15s`
+  // An already-fetched value should be refetched if it is older than the pollingInterval.
+  // Without explicit refetch, it would not be refetched until another pollingInterval has elapsed.
   const [trigger] = useLazyGetQuoteQuery({ pollingInterval })
   const request = useCallback(() => {
     const { refetch } = trigger(queryArgs, /*preferCacheValue=*/ true)
-    // An already-fetched value should be refetched if it is older than the pollingInterval.
-    // Without explicit refetch, it would not be refetched until another pollingInterval has elapsed.
-    if (fulfilledTimeStamp && Date.now() - fulfilledTimeStamp > pollingInterval) {
+    if (!isFetching && fulfilledTimeStamp && Date.now() - fulfilledTimeStamp > pollingInterval) {
       refetch()
     }
-  }, [fulfilledTimeStamp, pollingInterval, queryArgs, trigger])
+  }, [fulfilledTimeStamp, isFetching, pollingInterval, queryArgs, trigger])
   useTimeout(request, 200)
 
-  const [currencyIn, currencyOut] = isExactInput(tradeType)
-    ? [amountSpecified?.currency, otherCurrency]
-    : [otherCurrency, amountSpecified?.currency]
+  const result = useMemo(() => (typeof data === 'object' ? data : undefined), [data])
```

Comment:
is this check expensive enough that it needs to be memoized? don't quite understand this

Reply:
probably not, I'll remove it.
I consider any computation as needing memoization, to reduce computation time; or any logic which would return a referentially unstable value, to prevent unnecessary rerenders.